File: sbe/binance/spot_3_1/generated-python/spot_sbe/PriceTickerResponse.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class Tickers:
    """Repeating group item."""

    def __init__(self):
        self.price_exponent = None
        self.price = None

class PriceTickerResponse:
    """SBE message: PriceTickerResponse."""

    TEMPLATE_ID = 210
    SCHEMA_ID = 3
    SCHEMA_VERSION = 1
    BLOCK_LENGTH = 0

    def __init__(self):
        self.tickers = []

    def _decode_tickers_group(self, data: bytes, offset: int) -> Tuple[List[Tickers], int]:
        """Decode repeating group."""
        pos = offset

        block_length = struct.unpack_from('<H', data, pos)[0]
        pos += 2
        num_in_group = struct.unpack_from('<I', data, pos)[0]
        pos += 4

        items = []
        for _ in range(num_in_group):
            item_start = pos
            item = Tickers()

            item.price_exponent = struct.unpack_from('<b', data, pos)[0]
            pos += 1
            item.price = struct.unpack_from('<q', data, pos)[0]
            pos += 8

            # Skip to next block for forward compatibility
            pos = item_start + block_length
            items.append(item)

        return (items, pos)

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()


        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0


        # Skip to end of block for forward compatibility
        pos = 0

        self.tickers, pos = self._decode_tickers_group(data, pos)

File: sbe/binance/spot_3_1/generated-php/spot_sbe/PriceTickerResponse.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class Tickers
{
    public int|float|null $priceExponent = null;
    public int|float|null $price = null;
}

class PriceTickerResponse
{
    public const TEMPLATE_ID = 210;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 1;
    public const BLOCK_LENGTH = 0;

    public array $tickers = [];

    private function decodeTickersGroup(string $data, int &$offset): array
    {
        $blockLength = unpack('v', substr($data, $offset, 2))[1];
        $offset += 2;
        $numInGroup = unpack('V', substr($data, $offset, 4))[1];
        $offset += 4;

        $items = [];
        for ($i = 0; $i < $numInGroup; $i++) {
            $itemStart = $offset;
            $item = new Tickers();

            $item->priceExponent = unpack('c', substr($data, $offset, 1))[1];
            $offset += 1;
            $item->price = unpack('q', substr($data, $offset, 8))[1];
            $offset += 8;

            // Skip to next block for forward compatibility
            $offset = $itemStart + $blockLength;
            $items[] = $item;
        }

        return $items;
    }

    public function encode(): string
    {
        $buffer = '';


        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;


        // Skip to end of block for forward compatibility
        $offset = 0;

        $this->tickers = $this->decodeTickersGroup($data, $offset);
    }
}
